service/notification/NotificationProcessor.php

```diff
@@ -167,6 +179,10 @@ public function processNotification($unprocessedNotification)
                         $this->orderService->addPaymentDataToOrderFromResponse($order, $unprocessedNotification);
                     }
 
+                    if (!$this->validateWithCartAndOrder($unprocessedNotification, $order)) {
```

Comment:
this should be done before setting the order state to PS_OS_PAYMENT - in case it's set to PS_OS_PAYMENT at first then the customer is going to get an email that they payment was successful while after it's tagged as needs attention